### PR TITLE
WebGPURenderer: CubeRenderTarget and CubeCamera

### PR DESCRIPTION
**Pull request by sunag**
**Description**

This update is a continuation of this PR ( https://github.com/mrdoob/three.js/pull/26140 ) which includes the coordinate system for `WebGPU`. This approach allows `CubeCamera` to get the `.coordinateSystem` from the `Renderer` and update the settings if necessary. The update also makes some improvements to RTT regarding cache usage in case the application need to change many times the `Tone Mapping` and `Color Space`.

https://github.com/mrdoob/three.js/assets/502810/d6ba9332-7dbb-4fe5-871b-9cd50a95ddf2

- [x] Added CubeRenderTarget.fromEquirectangularTexture()
- [x] [`CubeCamera`: Introduce Coordinate System](https://github.com/mrdoob/three.js/commit/06e4c92fec90a552e3df7a3e204e06569d7bca24)
- [x] [WebGPURenderer: Added CubeRenderTarget support](https://github.com/mrdoob/three.js/commit/8d9cd59ca0b0fd82a5af772fe01b7196118524f6)
- [x] [NodeMaterial: Fixing currently colorSpace and environment](https://github.com/mrdoob/three.js/commit/283aacc78cf8b56f29156c058ddaeef40aa3281f)
- [x] [ToneMappingNode: Fix caching](https://github.com/mrdoob/three.js/commit/f5636b85df16b2cf24b1fe8d10723a9d56830e7e)
- [x] [Renderer Nodes: cleanup](https://github.com/mrdoob/three.js/commit/188d96bdaaf012dd87434354cfd211e1e0e3726b)
- [x] [Examples: Added webgpu_cubemap_dynamic](https://github.com/mrdoob/three.js/commit/0c6d059e27d3cbf48df918b88ae7d5cb459bd6ad)

### CODE REVIEW

**Comment by sunag** [comment]
@mrdoob I'm already very close to the `IBL` result expected in `WebGPU` for `Equirectangular Textures`.

| WebGPU  | WebGL |
| ------------- | ------------- |
| image | image |

## Textures

One thing has been bugging me for a while, it seems we have a texture configuration policy in three.js. Texture settings are often tweaked overriding the defined settings automatically. I'll give you an example:

https://github.com/mrdoob/three.js/blob/129635bdb5cbfa11f1b31522889846e862da43d0/examples/webgl_loader_gltf.html#L54-L61

image

### The settings are:
```js
texture.minFilter = THREE.LinearFilter;
texture.generateMipmaps = false;
//texture.needsPMREMUpdate = false;
```

### Texture settings are converted to
```js
texture.minFilter = THREE.LinearMipmapLinearFilter;
texture.generateMipmaps = true;
//texture.needsPMREMUpdate = true;
```

This is something I'm going to implement it texture configuration policy in the WebGPU to maintain backwards compatibility I think... I don't know if we can change that at this point, or if there is one description to this somewhere that I haven't noticed.

/cc @mrdoob @Mugen87 @LeviPesin

**Comment by mrdoob** [comment]
Is this the line you're referring to?

https://github.com/mrdoob/three.js/blob/129635bdb5cbfa11f1b31522889846e862da43d0/src/renderers/WebGLCubeRenderTarget.js#L123

File: examples/jsm/nodes/functions/PhysicalLightingModel.js [hunk 1]
```diff
@@ -40,7 +41,9 @@ const RE_IndirectSpecular_Physical = new ShaderNode( ( inputs ) => {
 
 	computeMultiscattering( singleScattering, multiScattering );
 
-	const diffuse = diffuseColor.mul( singleScattering.add( multiScattering ).oneMinus() );
+	const totalScattering = singleScattering.add( multiScattering );
+
+	const diffuse = diffuseColor.mul( max( totalScattering.r, max( totalScattering.g, totalScattering.b ) ).oneMinus() );
```
Comment (LeviPesin) [inline examples/jsm/nodes/functions/PhysicalLightingModel.js:46]:
```suggestion
	const diffuse = diffuseColor.mul( max( totalScattering.r, totalScattering.g, totalScattering.b ).oneMinus() );
```

Comment (sunag) [reply examples/jsm/nodes/functions/PhysicalLightingModel.js]: It was the first thing I did ( even deserves an update soon ), but I think we just added it to `OperatorNode`. I updated using chain approach for now.

**Pull request merged**